packages/core/src/combinator/multicast.js

```diff
@@ -0,0 +1,98 @@
+import { append, remove, findIndex } from '@most/prelude'
+import { tryEvent, tryEnd } from '../source/tryEvent'
+
+const dispose = disposable => disposable.dispose()
+
+const emptyDisposable = {
+  dispose () {}
+}
+
+class MulticastDisposable {
+  constructor (source, sink) {
+    this.source = source
+    this.sink = sink
+    this.disposed = false
+  }
+
+  dispose () {
+    if (this.disposed) {
+      return
+    }
+    this.disposed = true
+    const remaining = this.source.remove(this.sink)
+    return remaining === 0 && this.source._dispose()
```

Comment:
We should probably update `dispose` to return undefined that it's strict with types, e.g.

```js
    if(this.source.remove(this.sink) === 0) {
      this.source._dispose()
    }
```